software/contrib/piconacci.py

```diff
@@ -0,0 +1,145 @@
+from europi import *
+import machine
+from time import ticks_diff, ticks_ms
+from random import randint, uniform, choice
+
+from europi_script import EuroPiScript
+
+"""
+Piconacci
+author: Sean Bechhofer (github.com/seanbechhofer)
+date: 2022-05
+labels: triggers, maths
+
+Triggers based on the Fibonacci sequence
+
+digital_in: clock in
+analog_in: 
+
+knob_1: 
+knob_2: 
+
+button_1: Short Press: Move window on sequence left. Long Press: Rotate values left
+button_2: Short Press: Move window on sequence to the right. Long Press: Rotate values right
+output_1: trigger
+output_2: trigger
+output_3: trigger
+output_4: trigger
+output_5: trigger
+output_6: trigger
+
+"""
+
+
+# This probably works for most sensible circumstances. The 50th value
+# (disregarding 0, 1) is 20,365,011,074. At a BPM of 120, that'll be a
+# trigger every 300 years which is quite a lot even for slow moving
+# ambient pieces,
+MAX_FIB = 50
+
+
+class Piconacci(EuroPiScript):
+    def __init__(self):
+        # Flag to indicate whether the display needs updating.
+        self.dirty = True
+        # offset determines the sublist to be used for the values.
+        self.offset = 0
+        # rotate adjusts how the values are mapped to the cv
+        # outputs. A value of 0 means that cv1 is controlled by the
+        # first, cv2 the second etc. A rotate value of 1 means that
+        # cv2 is controlled by the first, cv3 the second, and cv1 the
+        # last.
+        self.rotate = 0
+        # Overclock the Pico for improved performance.
+        machine.freq(250_000_000)
+        # List of fibonacci numbers.
+        self.fib = [1, 1]
+        for i in range(0, MAX_FIB):
+            self.fib.append(self.fib[-2] + self.fib[-1])
+        # Strip off first 1
+        self.fib = self.fib[1:]
+
+        self.steps = [0] * 6
+
+        # Triggered when button 1 is released
+        # Short press:
+        # Long press:
```

Comment:
[optional] These three lines of comments don't seem to be adding value that isn't captured by the decorator. Same with the comments on `b2Pressed` below.